### DISCUSSION TITLE
Would like to contribute an s3 state module

### DISCUSSION THREAD

**Discussion by shivamMg** [saltstack/salt#67405] [Feature Requests]
### Description of Issue

[s3 module](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.s3.html) does not have a corresponding state module to make it easy for use in state files. I've managed to make one (partially as of now), and was wondering if you guys are interested in having it. I would love to contribute.

**Comment by waynew** [comment]
If there's no corresponding state, we'd love contributions 👍 

We require test coverage for new code, and you'd definitely want to sync up with the cloud working group - @Akm0d could get you hooked up there!

**Comment by shivamMg** [comment]
Would be great to have pointers there :+1:. Also, I'll use tests defined here https://github.com/saltstack/salt/blob/master/tests/unit/states as a reference.

**Comment by mchugh19** [comment]
@shivamMg documentation for writing state modules is here: https://docs.saltstack.com/en/latest/ref/states/writing.html

In short, generally states are wrappers for the functionality in modules. Since the s3 module exists, the state just needs to work idempotently, return a dict notifying about success/problems, and support `test=True`.

**Comment by shivamMg** [comment]
@mchugh19 Thank you. That should definitely help.
